test/e2e/branding_test.go

```diff
@@ -0,0 +1,198 @@
+
+	time.Sleep(1)
+
+	setOperatorConfigBrand(t, client, originalConfigBrand)
+
+	if originalManagedConfigMap.Name != "" {
+		originalManagedConfigMap.SetResourceVersion("0")
+	} else {
+		originalManagedConfigMap = createConsoleConfigMap(operatorsv1.BrandOKD)
+	}
+	err := client.ConfigMaps(consoleapi.OpenShiftConfigManagedNamespace).Delete(consoleapi.OpenShiftConsoleConfigMapName, &metav1.DeleteOptions{})
+	if err != nil {
+		t.Fatalf("could not delete managed config map %s, %v", originalManagedConfigMap.GetName(), err)
+	}
+	_, err = client.ConfigMaps(consoleapi.OpenShiftConfigManagedNamespace).Create(originalManagedConfigMap)
+	if err != nil {
+		t.Fatalf("could not reset managed config map %s, %v", originalManagedConfigMap.GetName(), err)
+	}
+
+}
+
+// Test setting brand using the operator config which overrides the default and managed configmap
+func TestOperatorConfigBranding(t *testing.T) {
+	client, originalConfigBrand, originalManagedConfigMap := setupBrandingTestCase(t)
+	defer cleanupBrandingTestCase(t, client, originalConfigBrand, originalManagedConfigMap)
+	// Set a temporary managed config to test it does not override the operator config values
+	_, err := updateOrCreateConsoleConfigMap(client, consoleapi.OpenShiftConfigManagedNamespace, createConsoleConfigMap(operatorsv1.BrandOKD))
+	if err != nil {
+		t.Fatalf("error: could not apply managed config map %v", err)
+	}
+
+	for _, expectedBrand := range validBrands {
+		t.Logf("update operator with %v", expectedBrand)
+		// helper to update the operator config
+		setOperatorConfigBrand(t, client, expectedBrand)
+		// now check if it has set the brand
+		err = wait.Poll(1*time.Second, pollTimeout, func() (stop bool, err error) {
+			gotBrand := getConsoleBrand(t, client)
+			//gotBrand := getOperaterConfigBrand(t, client)
+			return (gotBrand == expectedBrand), nil
+		})
+	}
+
+}
+
+// Test setting brand via the config map in the openshift-config-managed namespace, this requires the operator config not be set
+func TestBrandingFromManagedConfigMap(t *testing.T) {
+	client, originalConfigBrand, originalManagedConfigMap := setupBrandingTestCase(t)
+	defer cleanupBrandingTestCase(t, client, originalConfigBrand, originalManagedConfigMap)
+	// Set operator config to empty so it does not override the managed config map values
+	setOperatorConfigBrand(t, client, "")
+
+	for _, expectedBrand := range validBrands {
+		t.Logf("update data for the config map in openshift-config-managed namespace with %v", expectedBrand)
+		// Create new configmap for test
+		_, err := updateOrCreateConsoleConfigMap(client, consoleapi.OpenShiftConfigManagedNamespace, createConsoleConfigMap(expectedBrand))
+		if err != nil {
+			t.Fatalf("error: could not apply managed config map %v", err)
+		}
+
+		err = wait.Poll(1*time.Second, pollTimeout, func() (stop bool, err error) {
+			gotBrand := getConsoleBrand(t, client)
+			return (gotBrand == expectedBrand), nil
+		})
+	}
+}
+
```

Comment:
We could have a third if we wanted to:

```golang
// this e2e test validates that 
// - if brand is set on console-config in openshift-config-managed
// - and if brand is set on operator-config 
// - operator-config brand wins 
func TestBrandingFromManagedConfigAndOperatorConfig(t *testing.T) { 
	// create client 
	// get managed config 
	// if not found, create default 
	// get operator config 
	// set $brand on operator config 
	// get console-config in openshift-console 
	// check if console-config matches $brand 
	// throw error if incorrect
}
```

Reply:
This is what the current version of TestOperatorConfigBranding is doing.